pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -214,9 +214,20 @@ func ingest(ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.
 	progress := ingestionJob.Progress()
 	streamAddress := streamingccl.StreamAddress(details.StreamAddress)
 
-	startTime := progress.GetStreamIngest().StartTime
-	// Start from the last checkpoint if it exists.
+	// startTime is the timestamp from which the source cluster will begin
+	// streaming MVCC revisions. Consequently, it is the timestamp from which all
+	// partitions will begin ingesting data on the destination cluster.
+	// If there has been no checkpoint since the last resumption of the
+	// replication job then we use the timestamp that was picked during
+	// replication planning. Otherwise, we use the last checkpoint.
+	//
+	// Note, we only want the rangefeed on the source cluster to perform an
+	// initial scan if the replication job has reported no progress since the last
+	// resumption.
+	startTime := details.ReplicationStartTime
+	withInitialScan := true
```

Comment:
I don't love the idea of the destination having to tell the source whether or not to run an initial scan explicitly. It seems like a leakage across API boundaries IMO but it was the easiest to implement.

The alternative as @stevendanna mentioned was passing both the `ReplicationStartTime` and the `StartTime` where the latter changes across resumptions. Then the first time we setup the rangefeed we check if the `StartTime` is empty and use that as a signal whether we need to run an inital scan or not. Passing around two timestamps might be more confusing though? Open to thoughts.